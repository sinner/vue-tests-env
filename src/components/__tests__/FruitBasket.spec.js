import FruitBasket from '@/components/FruitBasket.vue';
import { mount } from '@vue/test-utils';

describe('Fruitbasket.vue', () => {
  it('can add fruits to basket with DOM', () => {
    const wrapper = mount(FruitBasket);

    const inputText = wrapper.find('input');
    const button = wrapper.find('button');

    inputText.element.value = 'banana';
    inputText.trigger('input'); // This line allows to trigger the event to vue could be able to change its state
    expect(wrapper.vm.fruit).toBe('banana');

    button.trigger('click');
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.fruit).toBe('');
      expect(wrapper.vm.basket).toEqual(expect.arrayContaining(['banana']));
      expect(wrapper.findAll('li').length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should call addToBasket when the button is clicked', () => {
    const wrapper = mount(FruitBasket);

    const addToBasketMuck = jest.fn().mockReturnValue('It was clicked');

    wrapper.setMethods({
      addToBasket: addToBasketMuck,
    });

    const inputText = wrapper.find('input');
    const button = wrapper.find('button');

    inputText.element.value = 'banana';
    inputText.trigger('input'); // This line allows to trigger the event to vue could be able to change its state

    button.trigger('click');
    wrapper.vm.$nextTick(() => {
      expect(addToBasketMuck).toHaveBeenCalled();
    });
  });
});
