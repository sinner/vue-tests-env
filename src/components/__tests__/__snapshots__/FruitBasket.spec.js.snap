// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fruitbasket.vue can add fruits to basket with DOM 1`] = `
<div>
  <h1>Fruit Basket</h1> <input type="text"> <button>Add Fruit</button>
  <ul class="fruits">
    <li>
      banana
    </li>
  </ul>
</div>
`;
